### ISSUE TITLE
Problems using shaderc_compile_options_set_include_callbacks

### ISSUE THREAD

**Issue by ramidzkh** [LWJGL/lwjgl3#585]
### Environment
* LWJGL version: `3.2.4-SNAPSHOT`
* Java version: `AdoptOpenJDK jdk-8.0.252.09-hotspot`
* Platform: `Windows`
* Module: `shaderc`

### Description
Trying to use a callback on `shaderc_compile_options_set_include_callbacks` always causes a segmentation fault and JVM crash (`0xC0000005`), even if the callback was run
```java
    public void setIncludeCallback(IncludeCallback callback) {
        assertOpen();

        shaderc_compile_options_set_include_callbacks(options,
                (user_data, requested_source, type, requesting_source, include_depth) -> {
                    IncludeResult result = callback.getResource(MemoryUtil.memUTF8Safe(requested_source), IncludeType.from(type), MemoryUtil.memUTF8Safe(requesting_source), include_depth);
                    return ShadercIncludeResult.calloc()
                            .set(MemoryUtil.memUTF8Safe(result.getPath()), MemoryUtil.memUTF8Safe(result.getContent()), user_data)
                            .address();
                },
                (user_data, include_result) -> {
                    System.out.println(user_data);
                    System.out.println(include_result);
                },
                MemoryUtil.NULL);
    }
```

### Reproducible sample
Run the `me.ramidzkh.compiler.Test` class in [test_case.zip](https://github.com/LWJGL/lwjgl3/files/5179007/test_case.zip) (I just realized I forgot to wrap the compiler in a try-with-resources but ignore that 🤦)

**Comment by ramidzkh** [comment]
I think I found the cause, https://github.com/LWJGL/lwjgl3/blob/master/modules/lwjgl/shaderc/src/generated/java/org/lwjgl/util/shaderc/ShadercIncludeResult.java#L321 unnecessarily checks if `user_data` is nullptr, which it is allowed to be

**Comment by httpdigest** [comment]
Until a next LWJGL version is built, you can workaround this by calling `ShaderIncludeResult.content(ByteBuffer)`, `ShaderIncludeResult.source_name(ByteBuffer)` and `ShaderIncludeResult.user_data(long)` (if needed) individually and omit calling `user_data` if it is NULL/0L.

**Comment by ramidzkh** [comment]
Thanks! Also, I managed to fix my `1 error: epic.hlsl:1: error: '' : unexpected token` issue by using `MemoryUtil.memUTF8(string, false)`. Would it be possible to make the documentation a little clearer about this?

**Comment by Spasi** [comment]
Thanks @ramidzkh! This has been fixed with 1ae4f5db346a0ab293b24141824b0885e109747b.
